osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -170,13 +162,24 @@ private void setExpansionStateOfGroup(GroupDefinition group, bool expanded)
         {
             if (grouping.GroupItems.TryGetValue(group, out var items))
             {
+                // First pass ignoring set groupings.
                 foreach (var i in items)
                 {
                     if (i.Model is GroupDefinition)
                         i.IsExpanded = expanded;
                     else
                         i.IsVisible = expanded;
                 }
+
+                // Second pass to hide set children when not meant to be displayed.
```

Comment:
Is there any reason this has to be a separate pass?

Reply:
It was just to simplify the logical application ordering (in the above loop it handles beatmaps that aren't grouped in a set, then we undo the incorrect ones).

I do foresee needing to change this for performance reasons. If you have an issue with it I can try that in this PR.

Reply:
> I do foresee needing to change this for performance reasons. If you have an issue with it I can try that in this PR.

I'm fine with letting you choose whether you want to address this now or later. The question was more to figure out the reasoning than anything.

Reply:
I'll fix this as part of this PR, so please hold off review until it's ready 😃